app/models/spree/taxon.rb

```diff
@@ -53,26 +53,29 @@ def pretty_name
 
     # Find all the taxons of supplied products for each enterprise, indexed by enterprise.
     # Format: {enterprise_id => [taxon_id, ...]}
-    def self.supplied_taxons
-      taxons = {}
-
-      Spree::Taxon.
+    #
+    # Optionally, specify some enterprise_ids to scope the results
+    def self.supplied_taxons(enterprise_ids = nil)
+      taxons = Spree::Taxon.
         joins(variants: :supplier).
-        select('spree_taxons.*, enterprises.id AS enterprise_id').
-        each do |t|
-          taxons[t.enterprise_id.to_i] ||= Set.new
-          taxons[t.enterprise_id.to_i] << t.id
-        end
+        select('spree_taxons.*, enterprises.id AS enterprise_id')
```

Comment:
It looks like we only need spree_taxons.id and enterprises.id here. Can this be optimised with a `pluck` also?

Reply:
Yes true! The only slight downside with it (compared to `select`) is that you then can't add any further scoping to it (`where`/`order` etc) because it returns an array then and there - but thankfully it's easy enough to rejig the order a bit here